.github/workflows/build.yaml

```diff
@@ -16,6 +16,7 @@ on:
         value: ${{ jobs.build.outputs.artifacts-sha256 }}
 permissions:
   contents: read
+  pull-requests: write
```

Comment:
We really want to avoid any `write` permissions in `build.yaml` for supply chain security. The reason is that it can build untrusted code from PRs. One thing we could do is to store the results of `make audit` as a job artifact and create a PR comment directly from `pull-request.yaml` in a job that has `write` permission.

Reply:
I'm working with this step! after doing the testing ill push all the fixes! thanks for the feedback and the explanation 😃